Firestore/core/src/firebase/firestore/util/async_queue.h

```diff
@@ -95,11 +96,30 @@ class AsyncQueue {
   // be called by a previously enqueued operation when it is run (as a special
   // case, destructors invoked when an enqueued operation has run and is being
   // destroyed may invoke `Enqueue`).
+  //
+  // After the shutdown process has initiated (`is_shutting_down()` is true),
+  // calling `Enqueue` will return, but the operation will *NOT* be run.
   void Enqueue(const Operation& operation);
 
+  // Like `Enqueue`, but also starts the shutdown process. Once the shutdown
+  // process has started, this queue will not run any operations requested
+  // via `Enqueue*` methods. They will not report error, they will simply
+  // return with the requested operations not scheduled.
+  //
+  // The exception is `EnqueueEvenAfterShutdown`, operations requsted via
+  // this will still be scheduled.
+  void EnqueueAndInitializeShutdown(const Operation& operation);
+
+  // Like `Enqueue`, but it will proceed scheduling the requested operation
+  // regardless if the queue is shut down or not.
```

Comment:
Nit: s/regardless if/regardless of whether.

Reply:
Done.